src/doc/contrib/src/process/working-on-cargo.md

```diff
@@ -46,6 +46,11 @@ If you can successfully run `cargo build`, you should be good to go!
 
 [homebrew]: https://brew.sh/
 
+Note: There is currently a bug on macOS that will cause an error while linking 
+cargo. To fix this, run `export MACOSX_DEPLOYMENT_TARGET=10.7` before attempting
```

Comment:
If this is going to be included in the contributing docs it may be worthwhile to word this a bit more precisely. I think that this only affects arm64 macOS machines, specifically M1 machines, so I don't think this needs to indicate that building on macOS is broken entirely.